### DISCUSSION TITLE
Are `spatialpandas`  and `antimeridian` optional dependencies?

### DISCUSSION THREAD

**Discussion by tomvothecoder** [UXARRAY/uxarray#756] [Q&A] [answered]
Hello, I'm going through the [Visualizing Grid Topology](https://uxarray.readthedocs.io/en/latest/examples/006-plot-api-topology.html) notebook in the documentation. 

I tried to run `grid.plot` but get `ModuleNotFoundError: No module named 'spatialpandas'`.

```python
File ~/miniforge3/envs/xcdat_test_uxarray/lib/python3.12/site-packages/uxarray/grid/geometry.py:134, in _build_geodataframe_with_antimeridian(polygon_shells, antimeridian_face_indices)
    131 """Builds a ``spatialpandas.GeoDataFrame`` including any faces that cross
    132 the antimeridian."""
    133 # import optional dependencies
--> 134 from spatialpandas.geometry import MultiPolygonArray
    135 from spatialpandas import GeoDataFrame
    137 polygons = _build_corrected_shapely_polygons(
    138     polygon_shells, antimeridian_face_indices
    139 )

ModuleNotFoundError: No module named 'spatialpandas'
```

I assumed that `uxarray` would include `spatialpandas` since the [Installation](https://uxarray.readthedocs.io/en/latest/installation.html) page doesn't list optional dependencies. However, I see that the [conda recipe](https://github.com/conda-forge/uxarray-feedstock/blob/main/recipe/meta.yaml) doesn't list `spatialpandas` and the stack trace says it is an optional dependency (line 133 above).

I'm just making sure if I should install `spatialpandas` alongside `uxarray`.  Thanks!

**Comment by tomvothecoder** [comment]
I installed `spatialpandas` and now get the same error for `antimeridian`.

```python
File ~/miniforge3/envs/xcdat_test_uxarray/lib/python3.12/site-packages/uxarray/grid/geometry.py:149, in _build_corrected_shapely_polygons(polygon_shells, antimeridian_face_indices)
    148 def _build_corrected_shapely_polygons(polygon_shells, antimeridian_face_indices):
--> 149     import antimeridian
    150     from shapely import polygons as Polygons
    152     # list of shapely Polygons representing each face in our grid

ModuleNotFoundError: No module named 'antimeridian'
```

**Comment by tomvothecoder** [comment]
I also see this GitHub issue reference the same things: https://github.com/UXARRAY/uxarray/issues/461#issue-1903470871

**Comment by philipc2** [comment, marked as answer]
Hi @tomvothecoder 

Both of those modules are indeed required. I'll look into getting the recipe fix. 

Thank you for catching this!

**Reply by tomvothecoder** [reply]
No problem! Thanks for the quick response @philipc2!
